public/src/theme.js

```diff
@@ -0,0 +1,50 @@
+import { css } from 'styled-components';
+
+export const theme = {
+  button: {
+    boxShadowSizes: '0px 0px 2px 2px',
+    transitionTiming: '.2s',
+    defaultStyling: css`
+      height: 40px;
+      cursor: pointer;
+      width: 100%;
+      display: flex;
+      align-items: center;
+      justify-content: center;
+      box-sizing: border-box;
+      text-align: center;
+      font-weight: bold;
+      padding: .25em 1em;
+      position: relative;
+      border-radius: .25em;
+      border-width: 2px;
+      border-style: solid;
+      & + & {
+        margin-top: .5em;
+      }
+      & + button {
+        margin-top: .5em;
+      }
+      & svg {
```

Comment:
These are generally a bad idea. What if you wanted a button with an svg on the right? The implementor should define the relationship of the elements.

Now, the spacing (.25em, .5em, 1em) is what the theme should define

Reply:
Hmm, how would you suggest I handle the spacing without the `{' '}` trick?

Reply:
Define the spacing relationship of the children in the...children...

```jsx
const IconButtonLabel = styled.span`
  ${({ theme }) => css`
   svg {
      margin-left: ${theme.margin('sm')}
   }
  `
```
Another possibility:

```jsx
const LeftIconButton = styled(Button)`
   & svg {
     margin-left: .25em; //replaced w/ theme
  }
`
```

or

```jsx
const ButtonIcon = styled(FontAwesomeIcon)`
     margin-left: .25em; //replaced w/ theme
`
```

Reply:
Done.